addons/stock/models/stock_orderpoint.py

```diff
@@ -339,37 +339,40 @@ def _search_qty_to_order(self, operator, value):
 
     @api.depends('qty_multiple', 'qty_forecast', 'product_min_qty', 'product_max_qty', 'visibility_days')
     def _compute_qty_to_order_computed(self):
-        for orderpoint in self:
-            if not orderpoint.product_id or not orderpoint.location_id:
-                orderpoint.qty_to_order_computed = False
-                continue
-            orderpoint.qty_to_order_computed = orderpoint._get_qty_to_order(qty_in_progress_by_orderpoint=orderpoint._quantity_in_progress())
+        def to_compute(orderpoint):
+            rounding = orderpoint.product_uom.rounding
+            # The check is on purpose. We only want to consider the visibility days if the forecast is negative and
+            # there is a already something to ressuply base on lead times.
+            return float_compare(orderpoint.qty_forecast, orderpoint.product_min_qty, precision_rounding=rounding) < 0
+
+        orderpoints = self.filtered(to_compute)
+        qty_in_progress_by_orderpoint = orderpoints._quantity_in_progress()
+        for orderpoint in orderpoints:
+            orderpoint.qty_to_order_computed = orderpoint._get_qty_to_order(qty_in_progress_by_orderpoint=qty_in_progress_by_orderpoint)
+        (self - orderpoints).qty_to_order_computed = False
 
-    def _get_qty_to_order(self, force_visibility_days=False, qty_in_progress_by_orderpoint={}):
+    def _get_qty_to_order(self, force_visibility_days=False, qty_in_progress_by_orderpoint=None):
         self.ensure_one()
-        if not self.product_id or not self.location_id:
-            return False
         visibility_days = self.visibility_days
         if force_visibility_days is not False:
             # Accepts falsy values such as 0.
             visibility_days = force_visibility_days
         qty_to_order = 0.0
+        qty_in_progress_by_orderpoint = qty_in_progress_by_orderpoint or {}
+        qty_in_progress = qty_in_progress_by_orderpoint.get(self.id)
+        if qty_in_progress is None:
+            qty_in_progress = self._quantity_in_progress()[self.id]
         rounding = self.product_uom.rounding
-        # The check is on purpose. We only want to consider the visibility days if the forecast is negative and
-        # there is a already something to ressuply base on lead times.
-        if float_compare(self.qty_forecast, self.product_min_qty, precision_rounding=rounding) < 0:
```

Comment:
I assume the removal of this check is related to the filtering done in `_compute_qty_to_order_computed`?
Sadly, this method could still be called from somewhere else, and thus still require that check to compute the right result.
And as a matter of fact, it's currently called in `action_product_forecast_report()`.

Reply:
Ah yeah nice catch